archaius2-api/src/main/java/com/netflix/archaius/api/PropertySource.java

```diff
@@ -0,0 +1,31 @@
+package com.netflix.archaius.api;
+
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+/**
+ * Contract for a raw source of properties
+ */
+public interface PropertySource {
+    /**
+     * Get the raw property value.  No interpolation or other modification is done to the property.
+     * @param key
+     */
+    default Optional<Object> getProperty(String key) { return Optional.empty(); }
+
+    /**
+     * Mechanism for consuming all properties of the PropertySource
+     * @param consumer
+     */
+    default void forEachProperty(BiConsumer<String, Object> consumer) {}
```

Comment:
Do we absolutely need this here? It doesn't match up with the Spring one, so just questioning if it really needs to be here or if its good enough on Config or some other sub-interface